[Winlogbeat] - TestWriteMaxUpdates is flaky

`TestWriteMaxUpdates` is flaky in Winlogbeat. I believe the cause is that the internal `run()` loop of checkpoint performs a `flush()` that conflicts with the test forcing a `read()` that is not synchronized. Normally checkpoint only does a `read()` once at startup so it isn't a problem in the application.

https://beats-ci.elastic.co/job/elastic+beats+master+multijob-windows/1330/beat=winlogbeat,label=windows/

```
=== RUN   TestWriteMaxUpdates
--- FAIL: TestWriteMaxUpdates (0.33s)
	checkpoint_test.go:47: predicate failed with error: open C:\Users\JENKIN~1.WOR\AppData\Local\Temp\wlb-checkpoint-test107915083\some\new\dir\.winlogbeat.yml: The process cannot access the file because it is being used by another process.
=== RUN   TestWriteTimedFlush
--- PASS: TestWriteTimedFlush (1.12s)
=== RUN   TestCreateDir
--- PASS: TestCreateDir (0.00s)
=== RUN   TestCreateDirAlreadyExists
--- PASS: TestCreateDirAlreadyExists (0.00s)
FAIL
coverage: 83.7% of statements
FAIL	github.com/elastic/beats/winlogbeat/checkpoint	1.505s
```

--- a/winlogbeat/checkpoint/checkpoint.go
+++ b/winlogbeat/checkpoint/checkpoint.go
@@ -40,6 +40,7 @@ type Checkpoint struct {
 	done          chan struct{}  // Channel for shutting down the checkpoint worker.
 	once          sync.Once      // Used to guarantee shutdown happens once.
 	file          string         // File where the state is persisted.
+	fileLock      sync.RWMutex   // Lock that protects concurrent reads/writes to file.
 	numUpdates    int            // Number of updates received since last persisting to disk.
 	maxUpdates    int            // Maximum number of updates to buffer before persisting to disk.
 	flushInterval time.Duration  // Maximum time interval that can pass before persisting to disk.
@@ -208,6 +209,9 @@ func (c *Checkpoint) persist() bool {
 
 // flush writes the current state to disk.
 func (c *Checkpoint) flush() error {
+	c.fileLock.Lock()
+	defer c.fileLock.Unlock()
+
 	tempFile := c.file + ".new"
 	file, err := create(tempFile)
 	if os.IsNotExist(err) {
@@ -258,6 +262,9 @@ func (c *Checkpoint) flush() error {
 // read loads the persisted state from disk. If the file does not exists then
 // the method returns nil and no error.
 func (c *Checkpoint) read() (*PersistedState, error) {
+	c.fileLock.RLock()
+	defer c.fileLock.RUnlock()
+
 	contents, err := ioutil.ReadFile(c.file)
 	if err != nil {
 		if os.IsNotExist(err) {

